### PR TITLE
FDUtils: Don't make unknown get_fdpath fatal

### PR DESCRIPTION
**Pull request by Sonicadvance1**
Encountered this while running wine things.
Non-fatal so don't explode

### CODE REVIEW

**Comment by skmp** [comment]
Several places in the codebase depend on being able to always get a string here. How about we convert this to `std::optional<std::string>` so we catch things correctly?

**Comment by Sonicadvance1** [comment]
Made it `std::optional<std::string>` and made the few places that use it ensure they are safely handling the case no value returns.

File: Source/Tests/LinuxSyscalls/FileManagement.cpp [hunk 1]
```diff
@@ -575,10 +575,14 @@ uint64_t FileManager::Readlinkat(int dirfd, const char *pathname, char *buf, siz
         dirfd != AT_FDCWD) {
     // Passed in a dirfd that isn't magic FDCWD
     // We need to get the path from the fd now
-    Path = FEX::get_fdpath(dirfd);
+    auto FDPath = FEX::get_fdpath(dirfd);
+
+    if (FDPath.has_value()) {
+      Path = FDPath.value();
+    }
```
Comment (neobrain) [inline Source/Tests/LinuxSyscalls/FileManagement.cpp:582]: FYI a nice way to simplify this is:
```suggestion
    auto Path = FEX::get_fdpath(dirfd).value_or("");
```

Not terribly important, just seemed worth mentioning :)

Comment (Sonicadvance1) [reply Source/Tests/LinuxSyscalls/FileManagement.cpp:582]: Guess that depends if we want the rest of the baggage with it, doesn't matter to me either way.
```diff
diff --git a/Source/Tests/LinuxSyscalls/EmulatedFiles/EmulatedFiles.cpp b/Source/Tests/LinuxSyscalls/EmulatedFiles/EmulatedFiles.cpp
index 1c053ac8f..a5916e43a 100644
--- a/Source/Tests/LinuxSyscalls/EmulatedFiles/EmulatedFiles.cpp
+++ b/Source/Tests/LinuxSyscalls/EmulatedFiles/EmulatedFiles.cpp
@@ -714,14 +714,10 @@ namespace FEX::EmulatedFile {
         dirfs != AT_FDCWD) {
       // Passed in a dirfd that isn't magic FDCWD
       // We need to get the path from the fd now
-      auto FDPath = FEX::get_fdpath(dirfs);
-
-      if (FDPath.has_value()) {
-        Path = FDPath.value();
-      }
+      Path = FEX::get_fdpath(dirfs).value_or("");

       if (pathname) {
-        if (FDPath.has_value()) {
+        if (!Path.empty()) {
           // If the path returned empty then we don't need a separator
           Path += "/";
         }
diff --git a/Source/Tests/LinuxSyscalls/FileManagement.cpp b/Source/Tests/LinuxSyscalls/FileManagement.cpp
index ce1bd7675..9b101fd23 100644
--- a/Source/Tests/LinuxSyscalls/FileManagement.cpp
+++ b/Source/Tests/LinuxSyscalls/FileManagement.cpp
@@ -575,14 +575,10 @@ uint64_t FileManager::Readlinkat(int dirfd, const char *pathname, char *buf, siz
         dirfd != AT_FDCWD) {
     // Passed in a dirfd that isn't magic FDCWD
     // We need to get the path from the fd now
-    auto FDPath = FEX::get_fdpath(dirfd);
-
-    if (FDPath.has_value()) {
-      Path = FDPath.value();
-    }
+    Path = FEX::get_fdpath(dirfd).value_or("");

     if (pathname) {
-      if (FDPath.has_value()) {
+      if (!Path.emtpy()) {
         // If the path returned empty then we don't need a separator
         Path += "/";
       }
       ```

Comment (neobrain) [reply Source/Tests/LinuxSyscalls/FileManagement.cpp:582]: Oh, I hadn't noticed we were actually using the `optional` after. Replacing the typesystem-level `has_value()` check with a value-level `empty()` isn't ideal, but I think I prefer it over managing the same state using two separate variables.

Comment (Sonicadvance1) [reply Source/Tests/LinuxSyscalls/FileManagement.cpp:582]: `I prefer it` Which one do you prefer? Original in PR or proposed diff in my previous comment? Ambiguous.

Comment (neobrain) [reply Source/Tests/LinuxSyscalls/FileManagement.cpp:582]: Apologies. I prefer the `value_or` approach, as it makes for shorter code overall and eliminates the need to have a dedicated variable for the `optional`.

Comment (Sonicadvance1) [reply Source/Tests/LinuxSyscalls/FileManagement.cpp]: blammo. Swapped over.

**Review by neobrain — APPROVED** [review]

**Comment by skmp** [comment]
:+1:

**Pull request merged**